openrtb3/openrtb3.go

```diff
@@ -0,0 +1,162 @@
+// Package openrtb3 holds OpenRTB 3.0 types implementation
+// (https://github.com/InteractiveAdvertisingBureau/openrtb)
+package openrtb3
+
+import "encoding/json"
+
+// Body is a top-level wrapper for OpenRTB requests.
+type Body struct {
+	OpenRTB OpenRTB `json:"openrtb"`
+}
+
+// ----------------------------------------------------------------------------
+
+// OpenRTB is the top-level object is the root for both request and response payloads.
+// It includes versioning information and references to the Layer-4 domain model on which transactions are based.
+// By default, the domain model used by OpenRTB is the Advertising Common Object Model (AdCOM).
+type OpenRTB struct {
+	Ver        string    `json:"ver,omitempty"`        // Version of the Layer-3 OpenRTB specification (e.g., "3.0").
+	DomainSpec string    `json:"domainspec,omitempty"` // Default: "adcom" (const DefaultOpenRTBDomainSpec). Identifier of the Layer-4 domain model used to define items for sale, media associated with bids, etc.
+	DomainVer  string    `json:"domainver"`            // Required. Specification version of the Layer-4 domain model referenced in the “domainspec” attribute.
+	Request    *Request  `json:"request,omitempty"`    // Required only for request payloads. Bid request container.
```

Comment:
Something to think about: less pointers - make specific Request/Response OpenRTB types (roughly, that copy `Ver`/`DomainSpec`/`DomainVer` but differ in having `Request`/`Response`).

Roughly, smth like:

```go
type OpenRTBBase struct {
  Ver, DomainSpec, DomainVer
}

type OpenRTBRequest struct {
  OpenRTBBase
  Request Request `json:"request"`
}

type OpenRTBResponse struct {
  OpenRTBBase
  Response Response `json:"response"`
}
```

Don't think that's really worth bothering though...

Anyway, may be done in future.